subtests/docker_cli/run_volumes_selinux/run_volumes_selinux.py

```diff
@@ -72,27 +75,33 @@ def init_container(self, volume=None, volumes_from=None):
         self.sub_stuff['fds'].append(write_fd)
         self.sub_stuff['fds'].append(read_fd)
         dkrcmd = dockercmd.AsyncDockerCmd(self, 'run', subargs)
-        # TODO: Fix use of dkrcmd.stdin when appropriate class mech. available
         dkrcmd.execute(read_fd)
         dkrcmd.stdin = write_fd
         os.close(read_fd)  # no longer needed
-        os.write(write_fd, 'echo "Started"\n')
-        self.failif(not wait_for_output(lambda: dkrcmd.stdout,
-                                        'Started',
-                                        timeout=20),
-                    "Error starting container %s: %s" % (name, dkrcmd))
+
+        # This could take a while if image needs pulling
+        find_name = lambda: str(dc.list_containers_with_name(name)) != []
+        self.failif_ne(utils.wait_for(find_name,
+                                      timeout=self.config['docker_timeout']),
+                       True,  # None == timeout
+                       "Container %s not in 'docker ps' output within"
+                       " timeout: %s" % (name, dc.list_containers()))
+
+        # Don't match echo of the 'echo' command itself
+        self.logdebug('Confirming %s is responding', name)
+        os.write(write_fd, "echo -e '%s'\n" % self.OCTAL_READY)
+        dkrcmd.wait_for_ready()
+        self.logdebug('Confirmed')
         return dkrcmd
 
     def check_context_recursive(self, path, context):
         """ Check all files in given $path have the context $context """
         for pwd, _, filenames in os.walk(path):
             for filename in filenames:
-                act = get_selinux_context("%s/%s" % (pwd, filename))
-                self.logdebug("File %s Context: %s",
-                              os.path.join(pwd, filename),
-                              act)
-                self.failif_ne(act, context, "Context of file %s/%s"
-                               % (pwd, filename))
+                actual = get_selinux_context("%s/%s" % (pwd, filename))
+                msg = ("File %s context: %s"
```

Comment:
Oooohhh that part.  Thanks, sorry, failed my Cptn. Obvious. Test today...twice... 😢